src/google/protobuf/message_lite.cc

```diff
@@ -554,6 +589,9 @@ struct ShutdownData {
   }
 
   static ShutdownData* get() {
+#ifdef _MSC_VER
+    MemoryIsNotDeallocated memory_is_not_deallocated;
```

Comment:
Does calling `ShutdownProtobufLibrary()` (see below) solve the problem? That function is intended to delete everything before your process exits if you want to be strict about that.

Reply:
@acozzette Thanks for the comments. We are publishing our library(both static and dynamic) to partner teams, and we can not ask them to call a function in protobuf which they do not depend on.

Reply:
Couldn't you provide your own cleanup function that calls `ShutdownProtobufLibrary()`? That way the projects that depend on your library wouldn't have to directly depend on protobuf.

Part of the reason I ask this is that if you don't call `ShutdownProtobufLibrary()`, I would expect there to be many other memory allocations that are never deleted. The `ShutdownData` object is just one heap-allocated object out of many.

Reply:
We currently do not provide cleanup function in our library. We can add that and communicate with our consumers to invoke that at the end, which would take some time and some of them may do not want to(may in the same situation as we are for their customers).

Currently ShutdownData object is the only one reported by CRT when application exits.

Reply:
Sorry but I don't think we want to accept this change. We call `OnShutdown`/`OnShutdownDelete`/`OnShutdownRun`/etc. in a bunch of places to register things that need to be destroyed or deleted at shutdown. It's possible that in your particular scenario you're not using any of the features involved and so perhaps in your case the `ShutdownData` object itself is the only thing that needs to be deleted, but in the general case this solution is not sufficient. In the general case, the only reliable way to account for all heap-allocated objects is to call `ShutdownProtobufLibrary()`.